### PR TITLE
Adding State Management Options to CLI

### PR DESCRIPTION
**Pull request by theblondealex**
## Description

I had added state management options to the ces flow
- Initially I have only added Zustand but with the approval of this PR I will look into adding other optons too
- Adds a question on the flow (see image) 
- Adds a --zustand flag to skip the question too.
This creates a folder in the root directory called state management with a default store from the zustand docs, it could be updated in future with a more generic store and some comments on how to use but to start with its the default store from zustand docs

## Motivation and Context

My motivation was as I believe that ces can and should become one of the best and number one go to option for starting new RN expo apps and state management is mandatory in almost every app asit scales and having the setup from the get go is a must, as well as this the intuitive nature of ces means that for beginner RN devs they can use it and this will ease them into state management, in short ces should have eveything you could need, state management is one such item

## Screenshots (if appropriate):

image

### CODE REVIEW

**Comment by danstepanov** [comment]
@theblondealex planning to review this for the next version of CES (next week)

**Comment by Avaray** [comment]
I really wish to see this in the CLI

**Comment by theblondealex** [comment]
Just updated it to resolve some conflicts with the recent main branch push

File: cli/src/utilities/runCLI.ts
Comment (theblondealex) [reply cli/src/utilities/runCLI.ts:418]: re the first comment about flag as there is another [PR ](https://github.com/roninoss/create-expo-stack/pull/417) up for redux might be good to keep as an option and allow it there

Apologies for such a delay got distrated with the day job, here is the video you requested

*few out of date package warnings but that will be solved with a rebase to main/merge* can rebase if you'd like but thought get approval on it first than to waste time if other changes need to be made


https://github.com/user-attachments/assets/30c95f01-57fc-4126-a2c9-78f98b808b02

**Comment by theblondealex** [comment]
> Have you tested this while also using 'react-native-mmkv' ? Having a hard time finding examples of this setup and even more so if including @TanStack-Query.

Those are separate packages to zustand and aren't exactly state managers so most likely wouldnt be in here IMO, this is the Redux/zustand type packages

**Comment by dannyhw** [comment]
Hey sorry that its taken a while but I will be looking to get things moving on the state management templates and will be taking a look soon.

File: cli/src/utilities/runCLI.ts
Comment (dannyhw) [reply cli/src/utilities/runCLI.ts:398]: agree, lets add both redux and zustand

File: cli/src/utilities/runCLI.ts
Comment (dannyhw) [inline cli/src/utilities/runCLI.ts:415]: not sure this internationalisation thing is doing anything here

**Comment by dannyhw** [comment]
There were some weird things that seem to have been messed up from merge conflicts probably, I think i managed to resolve them though

**Comment by theblondealex** [comment]
> There were some weird things that seem to have been messed up from merge conflicts probably, I think i managed to resolve them though

Awesome does anything else need to be done? Can I help in any way?

**Review by dannyhw — APPROVED** [review]

**Comment by dannyhw** [comment]
I think we're all good, just made a few more adjustments

**Comment by dannyhw** [comment]
@danstepanov your change request is blocking merge could you take a look, I think we can go ahead with this as is. I'll merge the redux changes in basically right after this is merged.

**Review by danstepanov — APPROVED** [review]
LGTM

**Comment by dannyhw** [comment]
@theblondealex we're having trouble with rebasing this branch, looks like the commit history got a bit messed up.

Can you try rebasing against main?

**Comment by theblondealex** [comment]
@dannyhw I have squashed the commits and rebased it against the latest, this inadvertedly closed the PR so reopened it now and it should be ready to merge

**Comment by dannyhw** [comment]
just ran everything locally and looks like we're good

**Pull request merged**
